File: Java/1-50/FirstMissingPositive.java
public class FirstMissingPositive {
    public int firstMissingPositive(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= 0 || nums[i] > nums.length) nums[i] = nums.length + 1;
        }
        for (int i = 0; i < nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            if (index == nums.length) continue;
            nums[index] = -Math.abs(nums[index]);
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) return i + 1;
        }
        return nums.length + 1;
    }
}

File: Python/1-50/FirstMissingPositive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for idx, num in enumerate(nums):
            if num <= 0 or num > len(nums):
                nums[idx] = len(nums) + 1
        for i in nums:
            i = abs(i) - 1
            if i < len(nums):
                nums[i] = -abs(nums[i])
        for idx, num in enumerate(nums):
            if num > 0:
                return idx + 1
        return len(nums) + 1
